infrastructure-formula/python/pillar/network.py

```diff
@@ -0,0 +1,161 @@
+"""
+Copyright (C) 2024 Georg Pfuetzenreuter <mail+opensuse@georg-pfuetzenreuter.net>
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <https://www.gnu.org/licenses/>.
+"""
+
+from logging import getLogger
+
+from yaml import safe_load
+
+root = '/srv/salt-git/pillar'
```

Comment:
That's the third time I read this hardcoded path ;-)

Would it make sense to define it in a single place, to read it from the salt config (with possible issues if you define more than one salt root) or even to make it configurable?

Reply:
I was thinking to add a configuration file for these libraries, however it will require additional states to write it.
Alternative it can be passed using function arguments. Unfortunately I do not know of a way to access the pillar without passing it from function arguments.

Reply:
However so far, the directory is the same in all our Salt environments, hence I think of it to be duplication to pass it in all function calls.

Reply:
That sounds like making it configurable is not easy, and - at least for now - not worth the effort. So let's keep it hardcoded until we have a good reason to make it configurable.